x/twap/logic_test.go

```diff
@@ -42,76 +43,287 @@ func newExpRecord(accum0, accum1 sdk.Dec) types.TwapRecord {
 	}
 }
 
-func TestRecordWithUpdatedAccumulators(t *testing.T) {
+func (s *TestSuite) TestNewTwapRecord() {
+	// prepare pool before test
+	poolId := s.PrepareBalancerPoolWithCoins(defaultTwoAssetCoins...)
+
+	tests := map[string]struct {
+		poolId        uint64
+		denom0        string
+		denom1        string
+		expectedErr   bool
+		expectedPanic bool
+	}{
+		"denom with lexicographical order": {
+			poolId,
+			denom1,
+			denom0,
+			false,
+			false,
+		},
+		"denom with non-lexicographical order": {
+			poolId,
+			denom0,
+			denom1,
+			false,
+			false,
+		},
+		"new record with same denom": {
+			poolId,
+			denom0,
+			denom0,
+			true,
+			false,
+		},
+		"non existent pool id": {
+			poolId + 1,
+			denom1,
+			denom0,
+			false,
+			true,
+		},
+	}
+	for name, test := range tests {
+		s.Run(name, func() {
+			if test.expectedPanic {
```

Comment:
We can replace this logic with `ConditionalPanic` https://github.com/osmosis-labs/osmosis/blob/main/app/apptesting/osmoassert/assertions.go#L14

Reply:
good catch on this, but I personally am not in favor of using conditional Panic for test cases simple like this, I think `conditionalPanic` is worth using when we have more complexity over testing. LMk what you think

Reply:
hmm, i was thinking more from code clarity + reusability of defined methods perspective. 

Also, i'm wondering should we only use `ConditionalPanic` for just complex panic testing? It's such a simple call that performs the exact same logic. I thought it was meant for simple testing like this